api-report/merge-tree.api.md

```diff
@@ -732,7 +732,7 @@ export class Marker extends BaseSegment implements ReferencePosition {
     readonly type = "Marker";
 }
 
-// @public (undocumented)
+// @public @deprecated (undocumented)
 export function matchProperties(a: PropertySet | undefined, b: PropertySet | undefined): boolean;
```

Comment:
some of this, like this one, are not being removed, same problem in the other PR/changeset

Reply:
Oh -- I thought we had to deprecate it before marking it `@internal`?

Reply:
It's good practice to do so: in theory things marked internal should be changeable in a minor version. It's still not great to do so with our current setup, since we don't hide such things in our package exports, so customers might be using them anyway.

But marking something internal should be a breaking change and IMO it's good to start doing the exercises now of deprecating ahead of time on main before marking internal, as it actually will immediately break existing users once we strip our internal APIs in the published .d.ts (and not just if we make further breaking changes in the same minor).

Reply:
Maybe the changeset could say 'will be removed / marked internal' if we want to be specific, but from the customer's perspective it doesn't really matter. they shouldn't be using it